### PR TITLE
added projects, ability to filter projects, added time tracker with a…

### PR DESCRIPTION
**Pull request by markeltskhadadze**
…bility to add/select working hours and project

### CODE REVIEW

File: src/components/Facts.vue [hunk 1]
```diff
@@ -3,10 +3,10 @@
     <div class="container" data-aos="fade-up">
       <header class="section-header">
         <h3>Facts</h3>
-        <p>
+        <!-- <p>
           Sed ut perspiciatis unde omnis iste natus error sit voluptatem
           accusantium doloremque
-        </p>
+        </p> -->
```
Comment (ArtemMakarchenko) [inline src/components/Facts.vue:9]: Remove

File: src/components/admin-panel/Projects.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  },
+  components: {
+    ProjectsFilters
+  },
+  methods: {
+    ...mapMutations(['setResult'])
+  },
+  computed: {
+    ...mapGetters(['searchProjectResult', 'searchClientFilterResult',
```
Comment (ArtemMakarchenko) [inline src/components/admin-panel/Projects.vue:33]: Remove useless code

File: src/components/admin-panel/Projects.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      'searchAccessFilterResult', 'resultFilter', 'projects',
+      'result'
+    ])
+  },
+  created () {
+    this.setResult(this.projects)
+  }
```
Comment (ArtemMakarchenko) [inline src/components/admin-panel/Projects.vue:40]: this.$store.setResult(this.projects)

File: src/components/admin-panel/TimeTrackerRecorder.vue [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      return result
+    },
+    sumTime () {
+      const startValue = this.checkStartTime.split(':')
+      const newStartArrValue = (+startValue[0] * 3600000) + (+startValue[1] * 60000) + (+startValue[2] * 1000)
+      const endValue = this.checkEndTime.split(':')
+      const newEndArrValue = (+endValue[0] * 3600000) + (+endValue[1] * 60000) + (+endValue[2] * 1000)
+      const resultComputed = newEndArrValue - newStartArrValue
+      const result = Math.floor(resultComputed / (1000 * 60 * 60)) + ':' + Math.floor(resultComputed / (1000 * 60)) % 60 + ':' + Math.floor(resultComputed / 1000) % 60
+      return result
```
Comment (ArtemMakarchenko) [inline src/components/admin-panel/TimeTrackerRecorder.vue:157]: The same

File: src/store/modules/admin-panel/index.js [hunk 5]
```diff
@@ -117,13 +208,50 @@ export default {
...
+    projects: state => state.projects,
+    clients: state => state.projects.map(client => client.client),
+    memberName: state => state.members.map(member => member.name),
+    executors: state => state.projects.map(user => user.executor),
+    active: state => state.projects.map(user => user.active).reduce((acc, active) => {
+      if (acc.includes(active)) {
+        return acc
+      }
+      return [...acc, active]
+    }, []),
```
Comment (ArtemMakarchenko) [inline src/store/modules/admin-panel/index.js:246]: return acc.includes(active) ? acc : [...acc, active]

File: src/components/Portfolio.vue [hunk 6]
```diff
@@ -28,17 +28,12 @@
...
-              <a
-                :href="item.image"
-                data-lightbox="portfolio"
-                data-title="App 1"
-                class="link-preview"
-                ><i class="bi bi-plus"></i
-              ></a>
+              <viewer>
+                <img :src="item.image" class="img-fluid" alt="" />
               <a :href="item.link" class="link-details" title="More Details"
```
Comment (ArtemMakarchenko) [inline src/components/Portfolio.vue:33]: indent

File: src/components/admin-panel/ListItem.vue [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+export default {
+  name: 'ListItem',
+  data () {
+    return {
+      timeStart: this.items.timeStart,
+      timeEnd: this.items.timeEnd,
+      total: this.items.totalTime
+    }
+  },
+  props: {
```
Comment (ArtemMakarchenko) [inline src/components/admin-panel/ListItem.vue:33]: Props should be above data

File: src/components/admin-panel/TimeTracker.vue [hunk 8]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <div>
+    <h1>Time Tracker</h1>
+    <TimeTrackerRecorder />
+    <div class="time-on-week">
+      <p>This Week</p>
+      <p>Week total:</p>
+    </div>
+    <div class="new-">
```
Comment (ArtemMakarchenko) [inline src/components/admin-panel/TimeTracker.vue:9]: What is it this class 'new-'?

File: src/store/modules/admin-panel/index.js [hunk 9]
```diff
@@ -40,6 +108,29 @@ export default {
...
+          state.result = state.result.filter(project => project.name.includes(data))
+        } else {
+          data.forEach(filterItem => {
+            if (filterItem.type === 'billing') state.result = state.result.filter(project => project.billing.includes(filterItem.value))
+            if (filterItem.type === 'client') state.result = state.result.filter(project => project.client.includes(filterItem.value))
+            if (filterItem.type === 'active') state.result = state.result.filter(project => project.active.includes(filterItem.value))
+            if (filterItem.type === 'executor') state.result = state.result.filter(project => project.executor.includes(filterItem.value))
+          })
+        }
+      } else {
```
Comment (ArtemMakarchenko) [inline src/store/modules/admin-panel/index.js:131]: You can remove this 'else'. You are doing the same in the first string of this method

File: src/store/modules/admin-panel/index.js [hunk 10]
```diff
@@ -117,13 +207,37 @@ export default {
           text: error.response.data.error || error.response.data.message
         })
       }
+    },
+    addNewItemList (context, payload) {
```
Comment (ArtemMakarchenko) [inline src/store/modules/admin-panel/index.js:211]: Should this function be asynchrony?

File: src/store/modules/admin-panel/index.js (hunk 5 above)
Comment (ArtemMakarchenko) [reply src/store/modules/admin-panel/index.js:235]: You can remove useless code. Example:
```
.reduce((acc, active) => acc.includes(active) ? acc : [...acc, active], []),
```

File: src/store/modules/admin-panel/index.js [hunk 11]
```diff
@@ -117,13 +207,37 @@ export default {
...
+    projects: state => state.projects,
+    clients: state => state.projects.map(client => client.client),
+    memberName: state => state.members.map(member => member.name),
+    executors: state => state.projects.map(user => user.executor),
+    active: state => state.projects.map(user => user.active).reduce((acc, active) => {
+      return acc.includes(active) ? acc : [...acc, active]
+    }, []),
+    billible: state => state.projects.map(user => user.billing).reduce((acc, billing) => {
+      return acc.includes(billing) ? acc : [...acc, billing]
+    }, []),
```
Comment (ArtemMakarchenko) [inline src/store/modules/admin-panel/index.js:238]: The same

**Review by seriyyah — APPROVED** [review]

**Pull request merged**